### ISSUE TITLE
[Truffle] Handle END keyword

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#4254]
### Example (`example.rb` file)

```ruby
puts "1"
END { puts "3" }
puts "2"
```

### Expected Behavior

```
1
2
3
```
### Actual Behavior
```
1
3
2
```
Found in `mkmf`: https://github.com/jruby/jruby/blob/master/lib/ruby/stdlib/mkmf.rb#L2721

There appear to be some more tests for this in `jt test mri ruby/test_beginendblock.rb`

This should also work when requiring, e.g.:
`one.rb`:
```
puts "1"
require "two"
puts "2"
```
`two.rb`:
```
END { puts "3" }
```
Should result in:
```
1
2
3
```

**Comment by bjfish** [comment]
The begin keyword also appears to have an issue:
```ruby
puts "2"
BEGIN { puts "1" }
puts "3"
```
Expected
```sh
1
2
3
```
Actual
```sh
2
3
```

**Comment by chrisseaton** [comment]
I can't reproduce the `BEGIN` case - can you verify?

**Comment by bjfish** [comment]
@chrisseaton the `BEGIN` case looks good to me now

**Comment by chrisseaton** [comment]
Fixed by 1c18c37a61d9f5b3f70b04a732991abe873c518e. Also see #4257.
